src/DataCollection.Shared/ViewModels/AppInfoViewModel.cs

```diff
@@ -0,0 +1,153 @@
+//  *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  *   See the License for the specific language governing permissions and
+//  *   limitations under the License.
+//  ******************************************************************************/
+
+using Esri.ArcGISRuntime.OpenSourceApps.DataCollection.Shared.Commands;
+using Esri.ArcGISRuntime.OpenSourceApps.DataCollection.Shared.Messengers;
+using Esri.ArcGISRuntime.OpenSourceApps.DataCollection.Shared.Properties;
+using System;
+using System.Diagnostics;
+using System.Reflection;
+using System.Windows.Input;
+
+namespace Esri.ArcGISRuntime.OpenSourceApps.DataCollection.Shared.ViewModels
+{
+    /// <summary>
+    /// ViewModel for the app's about page.
+    /// </summary>
+    public class AppInfoViewModel : BaseViewModel
+    {
+        /// <summary>
+        /// Defines the _appVersion.
+        /// </summary>
+        private string _appVersion;
+
+        /// <summary>
+        /// Defines the _runtimeVersion.
+        /// </summary>
+        private string _runtimeVersion;
+
+        /// <summary>
+        /// Defines the _launchLicenseInfoCommand.
+        /// </summary>
+        private ICommand _launchLicenseInfoCommand;
+
+        /// <summary>
+        /// Gets a value indicating if the version of ArcGIS Runtime should be shown in the UI.
+        /// </summary>
+        public bool ShowRuntimeVersion { get => Settings.Default.ShowRuntimeVersion ?? false; }
+
+        /// <summary>
+        /// Gets a value indicating if the version of the app should be shown in the UI.
+        /// </summary>
+        public bool ShowAppVersion { get => Settings.Default.ShowAppVersion ?? false; }
+
+        /// <summary>
+        /// Gets a value indicating whether a link to license info should be shown in the UI.
+        /// </summary>
+        public bool ShowLicenseInfo { get => Settings.Default.ShowLicenseInfo ?? false; }
+
+        /// <summary>
+        /// Gets a value indicating whether ShowAppInfo.
+        /// </summary>
+        public bool ShowAppInfo { get => ShowAppVersion || ShowRuntimeVersion || ShowLicenseInfo; }
+
+        /// <summary>
+        /// Gets a formatted string including the app's name, platform, and version.
+        /// </summary>
+        public string AppVersion
+        {
+            get
+            {
+                if (_appVersion == null)
+                {
+
+                    var appName = Resources.GetString("TitleBar_Name_App");
+                    var platform = "";
+#if NET_CORE
+                    platform = "WPF (.NET Core)";
+#elif WPF
+                    platform = "WPF (.NET Framework)";
+#elif NETFX_CORE
+                    platform = "UWP";
+#endif
+
+                    var version = Assembly.GetExecutingAssembly().GetName().Version.ToString();
+
+                    _appVersion = $"{appName}\n{platform}\n{version}";
+                }
+                return _appVersion;
+            }
+        }
```

Comment:
Just a thought here, feel free to discard based on my misunderstanding of the role of a view model. I would consider chopping these values up into three separate variables, allowing the view to determine how they are formatted rather than assuming a newline separation is wanted.